cmd/registry/scoring/score.go

```diff
@@ -0,0 +1,366 @@
+		}
+
+		// Check that the scoreValue is within min/max limits
+		configuredMin := definition.GetInteger().GetMinValue() // 0 if not set
+		configuredMax := definition.GetInteger().GetMaxValue() // 0 if not set
+		if value < configuredMin {
+			return nil, fmt.Errorf("evaluated score value(%d) cannot be less than the configured min_value (%d)", value, configuredMin)
+		}
+		if value > configuredMax {
+			return nil, fmt.Errorf("evaluated score value(%d) cannot be greater than the configured max_value (%d)", value, configuredMax)
+		}
+
+		// Populate Value field in Score proto
+		score.Value = &rpc.Score_IntegerValue{
+			IntegerValue: &rpc.IntegerValue{
+				Value:    value,
+				MinValue: configuredMin,
+				MaxValue: configuredMax,
+			},
+		}
+
+		// Populate the severity field according to Thresholds
+		for _, t := range definition.GetInteger().GetThresholds() {
+			if value >= t.GetRange().GetMin() && value <= t.GetRange().GetMax() {
+				score.Severity = t.GetSeverity()
+				break
+			}
+		}
+
+	case *rpc.ScoreDefinition_Percent:
+		// Score proto expects float32 type
+		value := float32(0)
+
+		// Convert scoreValue to appropriate type
+		// evaluateScoreExpression can return either a float or int value.
+		// Both are valid for an integer.
+		if intVal, ok := scoreValue.(int64); ok {
+			value = float32(intVal)
+		} else if floatVal, ok := scoreValue.(float64); ok {
+			value = float32(floatVal)
+		} else {
+			return nil, fmt.Errorf("failed typecheck for output: expected either int or float64 got %s", scoreValue)
+		}
+
+		// Check that the scoreValue is within min/max limits
+		if value < 0 {
+			return nil, fmt.Errorf("evaluated score value(%f) cannot be less than 0", value)
+		}
+		if value > 100 {
+			return nil, fmt.Errorf("evaluated score value(%f) cannot be greater than 100", value)
+		}
+
+		// Populate Value field in Score proto
+		score.Value = &rpc.Score_PercentValue{
+			PercentValue: &rpc.PercentValue{
+				Value: value,
+			},
+		}
+
+		// Populate the severity field according to Thresholds
+		for _, t := range definition.GetPercent().GetThresholds() {
+
+			if value >= float32(t.GetRange().GetMin()) && value <= float32(t.GetRange().GetMax()) {
+				score.Severity = t.GetSeverity()
+				break
+			}
+		}
+
+	case *rpc.ScoreDefinition_Boolean:
+		// Convert scoreValue to appropriate type
+		boolVal, ok := scoreValue.(bool)
+		if !ok {
+			return nil, fmt.Errorf("failed typecheck for output: expected bool")
+		}
+
+		displayValue := ""
+
+		if boolVal {
+			if configuredDisplay := definition.GetBoolean().GetDisplayTrue(); len(configuredDisplay) > 0 {
+				displayValue = configuredDisplay
+			} else {
+				displayValue = "true"
+			}
+		} else {
+			if configuredDisplay := definition.GetBoolean().GetDisplayFalse(); len(configuredDisplay) > 0 {
+				displayValue = configuredDisplay
+			} else {
+				displayValue = "false"
+			}
+		}
```

Comment:
I think this can be improved by flattening it out.

```
var displayValue string
if t := definition.GetBoolean().GetDisplayTrue(); boolVal && t != "" {
	displayValue = t
} else if f := definition.GetBoolean().GetDisplayFalse(); !boolVal && f != "" {
	displayValue = f
} else {
	displayValue = strconv.FormatBool(boolVal)
}
```

Reply:
Done. Thanks for the suggestion!